src/Features/Core/Portable/Copilot/ImplementationDetails.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.Copilot;
+
+internal sealed class ImplementationDetails
+{
+    public string? Message { get; init; }
+
+    public SyntaxNode? ReplacementNode { get; init; }
+}
```

Comment:
doc these.  my *expectation* is that this is an or-type.  one of these must ALWAYS be set.  and i would consider it a bug with the impl of the service if it either:

1. doesn't return an ImplDetails for a requested memberdeclnode
2. if didn't fill either hte message or replacementnode.  so either you get a replacement node.  or you get a message saying why.